src/plugins/upf/upf_pfcp_api.c

```diff
@@ -1065,9 +860,21 @@ handle_update_pdr (upf_session_t * sx, pfcp_update_pdr_t * update_pdr,
 	ASSERT (!pool_is_free_index (gtm->upf_apps, p[0]));
 	app = pool_elt_at_index (gtm->upf_apps, p[0]);
 	update->pdi.adr.application_id = p[0];
-	update->pdi.adr.db_id = upf_adf_get_adr_db (p[0]);
+	update->pdi.adr.db_id = upf_adf_get_adr_db (p[0], &acl);
 	update->pdi.adr.flags = app->flags;
 
+        if (!ISSET_BIT (pdr->pdi.grp.fields, PDI_SDF_FILTER))
+          {
+            vec_reset_length (update->pdi.acl);
+            if (vec_len(acl) > 0)
+              vec_alloc (update->pdi.acl, _vec_len(acl));
+          }
+
+        vec_foreach (acl_rule, acl)
+          {
+            vec_add1(update->pdi.acl, *acl_rule);
+          }
```

Comment:
this should also work
```suggestion
        vec_append(update->pdi.acl, acl);
```

Reply:
Fixed